opencog/nlp/sureal/SuRealPMCB.cc

```diff
@@ -426,17 +428,82 @@ bool SuRealPMCB::grounding(const std::map<Handle, Handle> &var_soln, const std::
         shrinked_soln[kv.first] = kv.second;
     }
 
-    // store the solution; all common InterpretationNode are solutions for this
-    // grounding, so store the solution for each InterpretationNode
-    for (auto n : qItprNode)
+    // if there are more than one common InterpretationNodes at this point,
+    // we should also have more than one SetLinks here, so examine them one by one
+    for (Handle hSetLink : qSolnSetLinks)
```

Comment:
These SetLinks are gathered earlier when we are retrieving the InterpretationNodes (L370), and it returns false if any of the pred_solns doesn't share a common InterpretationNode with other pred_solns (L395), so it seems reaching here means these SetLinks are the ones we want?